README.md

```diff
@@ -1,3 +1,22 @@
 # CA$H€R
 
-![](http://4.bp.blogspot.com/_XdP6Lp2ceqY/TEJuww2sk2I/AAAAAAAAWpY/EmWweRXahGM/s1600/tumblr_l55qcmER041qznd83o1_500.jpg)
+Casher is a Ruby script used by [Travis Build](https://github.com/travis-ci/travis-build) to fetch, create, and update caches. Casher fetches a cache if one exists and creates one if it doesn't. It uses md5deep to detect changes in an existing cache and updates the cache if a difference in the MD5 checksums are found. Casher's source is hosted and used directly from Github.
+
+![Comic about cash](http://4.bp.blogspot.com/_XdP6Lp2ceqY/TEJuww2sk2I/AAAAAAAAWpY/EmWweRXahGM/s1600/tumblr_l55qcmER041qznd83o1_500.jpg)
+
+## Where it fits in
```

Comment:
And summarize what it does instead of how it works in conjunction with another utility (`travis-build`) in particular.